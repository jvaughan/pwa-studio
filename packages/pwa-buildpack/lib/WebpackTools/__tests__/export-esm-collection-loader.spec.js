const ESMCollectionLoader = require('../loaders/export-esm-collection-loader');
const { runLoader } = require('../../TestHelpers/testWebpackLoader');

test('exports arrays of modules', async () => {
    expect(
        (await runLoader(ESMCollectionLoader, 'export default []', {
            query: [
                {
                    items: [
                        ['import indexZero from "index-zero"', 'indexZero'],
                        ['import { indexOne } from "./index-one"', 'indexOne']
                    ],
                    warnings: ['A warning.']
                }
            ]
        })).output
    ).toMatchSnapshot();
});

test('exports arrays of modules', async () => {
    expect(
        (await runLoader(ESMCollectionLoader, 'export default []', {
            query: [
                {
                    items: [
                        ['import indexZero from "index-zero"', 'indexZero'],
                        ['import { indexOne } from "./index-one"', 'indexOne']
                    ],
                    warnings: ['A warning.']
                }
            ]
        })).output
    ).toMatchSnapshot();
});

test('fails informatively if template module does not export the right kind of empty collection', async () => {
    const { context, output } = await runLoader(
        ESMCollectionLoader,
        'export default {}',
        {
            query: [
                {
                    items: [
                        ['import indexZero from "index-zero"', 'indexZero'],
                        ['import { indexOne } from "./index-one"', 'indexOne']
                    ],
                    warnings: ['A warning.']
                }
            ]
        }
    );
    expect(context.getCalls('emitError')).toMatchSnapshot();
    expect(output).toBe('export default {}');
});

test('warns if nothing was added', async () => {
    const { context, output } = await runLoader(
        ESMCollectionLoader,
        'export default []',
        {
            query: [
                {
                    items: [],
                    warnings: []
                }
            ]
        }
    );
    expect(context.getCalls('emitWarning')).toMatchSnapshot();
    expect(output).toBe('export default []');
});
