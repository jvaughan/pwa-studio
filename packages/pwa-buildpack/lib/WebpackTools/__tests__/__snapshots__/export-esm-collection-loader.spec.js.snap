// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exports arrays of modules 1`] = `
"// generated by export-esm-collection-loader
import indexZero from \\"index-zero\\"
import { indexOne } from \\"./index-one\\"

export default [
  indexZero,
  indexOne
];"
`;

exports[`exports arrays of modules 2`] = `
"// generated by export-esm-collection-loader
import indexZero from \\"index-zero\\"
import { indexOne } from \\"./index-one\\"

export default [
  indexZero,
  indexOne
];"
`;

exports[`fails informatively if template module does not export the right kind of empty collection 1`] = `
Array [
  Array [
    [Error: items provided as array, but cannot export an array because template module "/content-file" does not export an empty array.
            Note that comments are not allowed between the [ and ] for these template modules, but should be placed above the export declaration.

            Contents of "/content-file":
            export default {}],
  ],
]
`;

exports[`warns if nothing was added 1`] = `
Array [
  Array [
    [Error: Items array was empty; no changes will be made],
  ],
]
`;
